CONTRIBUTING.md

```diff
@@ -100,18 +100,34 @@ checks, so please check back after a few minutes.
 
 Tests
 -----
-If your PR is a feature (ie. if it's being labelled with the 'T-Enhancement'
-label) then we require that the PR also includes tests. These need to test that
-your feature works as expected and ideally test edge cases too. For the js-sdk
-itself, your tests should generally be unit tests. matrix-react-sdk also uses
-these guidelines, so for that your tests can be unit tests using
-react-test-utils, snapshot tests or screenshot tests.
-
-We don't require tests for bug fixes (T-Defect) but strongly encourage regression
-tests for the bug itself wherever possible.
-
-In the future we may formalise this more with a minimum test coverage
-percentage for the diff.
+Your PR should include tests.
+
+For new user facing features in `matrix-react-sdk` or `element-web`, you
+must include:
```

Comment:
does this include features behind labs? Not sure where the clause went, but previously we had a (potentially informal) policy that tests had to be written for a feature to leave labs, but not to enter labs. I'd like to keep this clause in place.

Reply:
We can discuss, but we want to shoot for writing unit tests alongside new work, not after the fact; this means the code is much more likely to be testable and you get the benefit of having the tests there to make sure you're not breaking stuff as you go.

For e2e tests, I can see the argument for adding them at the end of feature development.

Reply:
The previous policy @turt2live mentions is described in https://github.com/vector-im/element-web/blob/develop/docs/review.md#code-quality, which I guess partially overlaps with this file (and perhaps conflicts with it).

Not sure which repo should be the canonical home of team process docs, but Element Web seems a tad more logical to me... 🤷 There are other useful tidbits in https://github.com/vector-im/element-web/blob/develop/docs/review.md, so it would seem best to preserve and update it where needed.

Reply:
Thanks. That policy isn't linked to from the element-web README afaict, which links to the develop guide for matrix-react-sdk, which links to this file.

Reply:
Yes, but all of them are documents we can edit... 😜 

Perhaps good to add separate document clean up task. At the moment, there are enough separate documents in different repos it's likely overwhelming for someone new (contributor or Element employee alike), especially when there are several covering similar topics with different text.